README.md

```diff
@@ -208,6 +208,24 @@ const swaggerDocument = YAML.load('./swagger.yaml');
 app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocument));
 ```
 
+
+### Modify swagger file on the fly before load
+
+To dynamically set the host, or any other content, in the swagger file based on the incoming request object you may pass the json via the req object; to achieve this just do not pass the the swagger json to the setup function and it will look for `swaggerDoc` in the `req` object.
+
+```javascript
+const express = require('express');
+const app = express();
+const swaggerUi = require('swagger-ui-express');
+app.use('/api-docs', function(req, res, next){
+    const swaggerDocument = require('./swagger.json');
```

Comment:
can you place this require outside the ```app.use```? don't want to promote blocking the thread